src/coreclr/jit/codegenwasm.cpp

```diff
@@ -164,6 +178,123 @@ void CodeGen::genCodeForBinary(GenTreeOp* treeNode)
         case PackOperAndType(GT_ADD, TYP_DOUBLE):
             ins = INS_f64_add;
             break;
+
+        case PackOperAndType(GT_SUB, TYP_INT):
+            ins = INS_i32_sub;
+            break;
+        case PackOperAndType(GT_SUB, TYP_LONG):
+            ins = INS_i64_sub;
+            break;
+        case PackOperAndType(GT_SUB, TYP_FLOAT):
+            ins = INS_f32_sub;
+            break;
+        case PackOperAndType(GT_SUB, TYP_DOUBLE):
+            ins = INS_f64_sub;
+            break;
+
+        case PackOperAndType(GT_MUL, TYP_INT):
+            ins = INS_i32_mul;
+            break;
+        case PackOperAndType(GT_MUL, TYP_LONG):
+            ins = INS_i64_mul;
+            break;
+        case PackOperAndType(GT_MUL, TYP_FLOAT):
+            ins = INS_f32_mul;
+            break;
+        case PackOperAndType(GT_MUL, TYP_DOUBLE):
+            ins = INS_f64_mul;
+            break;
+
+        case PackOperAndType(GT_DIV, TYP_INT):
+            ins = INS_i32_div_s;
+            break;
+        case PackOperAndType(GT_DIV, TYP_LONG):
+            ins = INS_i64_div_s;
+            break;
+        case PackOperAndType(GT_DIV, TYP_FLOAT):
+            ins = INS_f32_div;
+            break;
+        case PackOperAndType(GT_DIV, TYP_DOUBLE):
+            ins = INS_f64_div;
+            break;
+
+        case PackOperAndType(GT_UDIV, TYP_INT):
+            ins = INS_i32_div_u;
+            break;
+        case PackOperAndType(GT_UDIV, TYP_LONG):
+            ins = INS_i64_div_u;
+            break;
+
+        case PackOperAndType(GT_MOD, TYP_INT):
+            ins = INS_i32_rem_s;
+            break;
+        case PackOperAndType(GT_MOD, TYP_LONG):
+            ins = INS_i64_rem_s;
+            break;
+
+        case PackOperAndType(GT_UMOD, TYP_INT):
+            ins = INS_i32_rem_u;
+            break;
+        case PackOperAndType(GT_UMOD, TYP_LONG):
+            ins = INS_i64_rem_u;
+            break;
+
+        case PackOperAndType(GT_AND, TYP_INT):
+            ins = INS_i32_and;
+            break;
+        case PackOperAndType(GT_AND, TYP_LONG):
+            ins = INS_i64_and;
+            break;
+
+        case PackOperAndType(GT_OR, TYP_INT):
+            ins = INS_i32_or;
+            break;
+        case PackOperAndType(GT_OR, TYP_LONG):
+            ins = INS_i64_or;
+            break;
+
+        case PackOperAndType(GT_XOR, TYP_INT):
+            ins = INS_i32_xor;
+            break;
+        case PackOperAndType(GT_XOR, TYP_LONG):
+            ins = INS_i64_xor;
+            break;
+
+        case PackOperAndType(GT_LSH, TYP_INT):
```

Comment:
The shift operand width in IR is always `int`; the WASM operations have the same widths for both the shift and shiftee. So the shift may need to be extended (zero-extended) for `TYP_LONG`. It is probably best done in a separate `genCodeForShift`.

I don't remember the exact rules for rotates. You'll need to check them manually.